source/euler/hyperbolic_system.h

```diff
@@ -1313,5 +1339,159 @@ namespace ryujin
       return result;
     }
 
+
+    template<int dim, typename Number>
+    void HyperbolicSystem::View<dim, Number>::post_process(const vector_type &U,
+                                                           Number t,
+                                                           MPI_Comm mpi_communicator_,
+                                                           const ryujin::OfflineData<dim, Number>& offline_data_)
+    const
+    {
+    if(dim == 2){
+      using scalar_type = dealii::LinearAlgebra::distributed::Vector<Number>;
+
+
+       //some general variables
+       dealii::Point<dim> disk_center;//center at 0,0 for every case, how to not hardcode this?
+       //FIXME: how do we calculate disk center from the information in mesh generation?
+       double disk_radius = 0.25;
+
+       //first, set up the finite element, the data, and the facevalues
+       dealii::FE_Q<dim> fe(ORDER_FINITE_ELEMENT);//the finite element
+       const int degree = fe.degree;
+       dealii::QGauss<dim-1> face_quadrature_formula(degree + 2);
+       const int n_q_points = face_quadrature_formula.size();
+
+       std::vector<double>      pressure_values(n_q_points);
+
+       scalar_type density, pressure;
+       std::vector<scalar_type> momentum(dim);
+
+   //    std::cout << "inside calc before density partition" << std::endl;
+       //initialize partitions
+       density.reinit(offline_data_.scalar_partitioner(), mpi_communicator_);
+       pressure.reinit(offline_data_.scalar_partitioner(), mpi_communicator_);
+       for(unsigned int c=0; c < dim; c++)
+         momentum.at(c).reinit(offline_data_.scalar_partitioner(), mpi_communicator_);
+
+       dealii::Tensor<1,dim> normal_vector;
+       dealii::SymmetricTensor<2,dim> fluid_stress;
+       dealii::SymmetricTensor<2,dim> fluid_pressure;
+       dealii::Tensor<1,dim> forces;
+
+       dealii::FEFaceValues<dim> fe_face_values(fe, face_quadrature_formula,
+           dealii::update_values | dealii::update_quadrature_points | dealii::update_gradients |
+           dealii::update_JxW_values | dealii::update_normal_vectors); //the face values
+
+       double drag = 0.;
+       double lift = 0.;
+
+
+       // Create vectors that store the locally owned parts on every process
+       U.extract_component(density, 0);//extract density
+       U.extract_component(pressure, dim+1);//extract density
+
+       //extract momentum, and convert to velocity
+       for(int c=0; c<dim; c++)
+       {
+         int comp = c+1;//momentum is stored in positions [1,...,dim], so add one to c
```

Comment:
Is this true for all `HyperbolicSystem` classes? This seems specific to Euler-type problems, but is this class only doing that or handling more general hyperbolic systems as its name suggest?

Reply:
This is something I am going to bring up. Yes, this is certainly unique to Euler-type problems, and yes, this class is only handling the euler type equations, but also not... I'll explain the organization of ryujin in our meeting, but each type of problem type has its own subfolder wherein a separate, but structurally equivalent hyperbolic system template class is defined, then instantiated with the proper description. 

This is why, for some of the places in this code I added a postprocess() to a separate hyperbolic system file, and made it simply do nothing. Awkward I admit, and something I'd like to fix at some point.